lib/user_space/utrace.c

```diff
@@ -80,7 +84,8 @@ static int logbuf_map()
 		int available_bytes = sizeof trace_file_path -
 				      strlen(trace_file_path);
 		rc = snprintf(trace_file_path + strlen(trace_file_path),
-			available_bytes, "m0trace.%u", m0_pid_cached);
+			available_bytes, "m0trace.%u.%"PRIu32, m0_pid_cached,
+                                          (uint32_t )m0_time_now() & UTRACE_HIGH_32BIT);
```

Comment:
The value from m0_time_now() is already cast to uint32_t , then why do we need to apply the mask?

Reply:
m0_time_now returns uint64_t, right ?

m0_time_t m0_time_now(void)
lib/time.h::typedef uint64_t m0_time_t;

Reply:
Sorry, I meant to ask that we are truncating the value to uint32 and using it. So why do we need to apply the mask?

Reply:
You may need this to get the lower 32bits:
```
(uint32_t )(m0_time_now() & UTRACE_HIGH_32BIT)
```

But even with this, we are getting the lower 32bits, not the high 32bits.